accounts/queries.py

```diff
@@ -114,39 +114,37 @@ def get_child_participation_eligibility(child, study) -> bool:
     Returns:
         bool: Return true if child is eligible based on their prior study participation
     """
-    # Need to import StudyType here due to circular import problems
-    from studies.models import StudyType
 
-    must_have_participated = True
-    must_not_have_participated = True
+    ember_frame_player_id = 1
+    must_have = True
+    must_not = True
 
     # for both must have and must not have participated, ignore responses from internal studies that are empty
+
     if study.must_have_participated.exists():
-        must_have_participated_all = [
-            i
-            for i in study.must_have_participated.all()
-            if child.responses.filter(study=i)
-            .exclude(study__study_type=StudyType.get_ember_frame_player(), sequence=[])
-            .exists()
-        ]
-
-        # only true if response exists for each of the studies in the list
-        must_have_participated = set(must_have_participated_all) == set(
-            study.must_have_participated.all()
+        must_have_count = (
+            child.responses.filter(study__in=study.must_have_participated.all())
+            .exclude(study__study_type_id=ember_frame_player_id, sequence=[])
+            .distinct()
+            .values_list("study")
+            .count()
         )
+        must_have = must_have_count == study.must_have_participated.count()
+
```

Comment:
Along the same lines, I'm wondering just in case if there are other reasons I'm not thinking of why the number of studies might not match counts with the number of exclusions?  How much slower would it be to sort and ask if the two lists are identical, `study.must_have_participated.count()`  and the list generated in the 2nd to last line of that query?

Reply:
We can, it's just unnecessary.  

The responses for this query will be from a study in the `must_have_participated` list.  This is from the filter `study__in=study.must_have_participated.all()`.  

Here's SQL generated by `child.responses.filter(study__in=study.must_have_participated.all())`:

```SQL
select
	"studies_response"."id",
	"studies_response"."uuid",
	"studies_response"."study_id",
	"studies_response"."completed",
	"studies_response"."completed_consent_frame",
	"studies_response"."survey_consent",
	"studies_response"."exp_data",
	"studies_response"."conditions",
	"studies_response"."sequence",
	"studies_response"."date_modified",
	"studies_response"."date_created",
	"studies_response"."global_event_timings",
	"studies_response"."child_id",
	"studies_response"."is_preview",
	"studies_response"."demographic_snapshot_id",
	"studies_response"."study_type_id",
	"studies_response"."recording_method",
	"studies_response"."eligibility"
from
	"studies_response"
where
	("studies_response"."child_id" = 10176
		and "studies_response"."study_id" in (
		select
			U0."id"
		from
			"studies_study" U0
		inner join "studies_study_must_have_participated" U1 on
			(U0."id" = U1."to_study_id")
		where
			U1."from_study_id" = 3129))
```